document/core/exec/instructions.rst

```diff
@@ -2351,65 +2351,67 @@ Memory Instructions
 
 5. Let :math:`\X{mem}` be the :ref:`memory instance <syntax-meminst>` :math:`S.\SMEMS[\X{ma}]`.
 
-6. Assert: due to :ref:`validation <valid-memory.copy>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+6. Let :math:`\X{it}~\limits` be the :math:`\X{mem}.\MITYPE`.
 
-7. Pop the value :math:`\I32.\CONST~n` from the stack.
+7. Assert: due to :ref:`validation <valid-memory.copy>`, a value of :ref:`value type <syntax-valtype>` :math:`\X{it}` is on the top of the stack.
 
-8. Assert: due to :ref:`validation <valid-memory.copy>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+8. Pop the value :math:`\X{it}.\CONST~n` from the stack.
 
-9. Pop the value :math:`\I32.\CONST~s` from the stack.
+9. Assert: due to :ref:`validation <valid-memory.copy>`, a value of :ref:`value type <syntax-valtype>` :math:`\X{it}` is on the top of the stack.
 
-10. Assert: due to :ref:`validation <valid-memory.copy>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+10. Pop the value :math:`\X{it}.\CONST~s` from the stack.
 
-11. Pop the value :math:`\I32.\CONST~d` from the stack.
+11. Assert: due to :ref:`validation <valid-memory.copy>`, a value of :ref:`value type <syntax-valtype>` :math:`\X{it}` is on the top of the stack.
+
+12. Pop the value :math:`\X{it}.\CONST~d` from the stack.
 
-12. If :math:`s + n` is larger than the length of :math:`\X{mem}.\MIDATA` or :math:`d + n` is larger than the length of :math:`\X{mem}.\MIDATA`, then:
+13. If :math:`s + n` is larger than the length of :math:`\X{mem}.\MIDATA` or :math:`d + n` is larger than the length of :math:`\X{mem}.\MIDATA`, then:
 
     a. Trap.
 
-13. If :math:`n = 0`, then:
+14. If :math:`n = 0`, then:
 
    a. Return.
 
-14. If :math:`d \leq s`, then:
+15. If :math:`d \leq s`, then:
 
-   a. Push the value :math:`\I32.\CONST~d` to the stack.
+   a. Push the value :math:`\X{it}.\CONST~d` to the stack.
```

Comment:
I'm not sure if I can use `it` directly here or if I have to map it somehow?

Reply:
That's totally fine. But note that we now have multiple memories, so there will be two different it's.